scheduler/workflow_test.go

```diff
@@ -121,3 +125,172 @@ func TestCollectPublishWorkflow(t *testing.T) {
 		})
 	})
 }
+
+// The mocks below are here for testing work submission
+type Mock1 struct {
+	sync.Mutex
+	count      int
+	errorIndex int
+	delay      time.Duration
+	queue      map[string]int
+}
+
+func (m *Mock1) CollectMetrics([]core.Metric, time.Time, string) ([]core.Metric, []error) {
+	return nil, nil
+}
+
+func (m *Mock1) Work(j job) queuedJob {
+	m.Lock()
+	defer m.Unlock()
+	time.Sleep(m.delay)
```

Comment:
I was using this test as a way of being sure the behavior I wanted was happening when queue time (before work) was greater than 0. This is left over when I converted the test harness into a test. I can remove this since I am not sending a delay into the Mock1 struct to remove confusion.